packages/core/src/components/date-picker-rework/date-picker-rework.tsx

```diff
@@ -0,0 +1,634 @@
+    const selectedDayObj = dayjs(
+      new Date(this.selectedYear, this.selectedMonth, day)
+    );
+
+    return {
+      'calendar-item': true,
+      'empty-day': day === undefined,
+      today: todayObj.isSame(selectedDayObj, 'day'),
+      selected:
+        this.currFromDate?.isSame(selectedDayObj, 'day') ||
+        this.currToDate?.isSame(selectedDayObj, 'day'),
+      range:
+        selectedDayObj.isAfter(this.currFromDate, 'day') &&
+        this.currToDate !== undefined &&
+        selectedDayObj.isBefore(this.currToDate, 'day'),
+      disabled: !this.isWithinMinMaxDate(selectedDayObj),
+    };
+  }
+
+  private isWithinMinMaxYear(year: number): boolean {
+    const minDateYear = this.minDate
+      ? dayjs(this.minDate, this.format, true).year()
+      : undefined;
+    const maxDateYear = this.maxDate
+      ? dayjs(this.maxDate, this.format, true).year()
+      : undefined;
+    const isBefore = minDateYear ? year < minDateYear : false;
+    const isAfter = maxDateYear ? year > maxDateYear : false;
+
+    return !isBefore && !isAfter;
+  }
+
+  private isWithinMinMaxMonth(month: number): boolean {
+    const minDateObj = this.minDate
+      ? dayjs(this.minDate, this.format, true)
+      : undefined;
+    const maxDateObj = this.maxDate
+      ? dayjs(this.maxDate, this.format, true)
+      : undefined;
+    const minDateMonth = minDateObj?.month();
+    const maxDateMonth = maxDateObj?.month();
+    const isBefore = minDateMonth
+      ? this.tempYear === minDateObj.year() && month < minDateMonth
+      : false;
+    const isAfter = maxDateMonth
+      ? this.tempYear === maxDateObj.year() && month > maxDateMonth
+      : false;
+
+    return !isBefore && !isAfter;
+  }
+
+  private isWithinMinMaxDate(date: Dayjs): boolean {
+    const _minDate = this.minDate
+      ? dayjs(this.minDate, this.format, true)
+      : undefined;
+    const _maxDate = this.maxDate
+      ? dayjs(this.maxDate, this.format, true)
+      : undefined;
+    const isBefore = _minDate ? date.isBefore(_minDate, 'day') : false;
+    const isAfter = _maxDate ? date.isAfter(_maxDate, 'day') : false;
+
+    return !isBefore && !isAfter;
+  }
+
+  private renderYears(): any[] {
+    const rows = [];
+
+    for (let year = this.startYear; year <= this.endYear; year++) {
+      rows.push(
+        <div
+          key={year}
+          class={{
+            arrowYear: true,
+            'disabled-item': !this.isWithinMinMaxYear(year),
+          }}
+          onClick={(event) => this.selectTempYear(event, year)}
+        >
+          <ix-icon
+            class={{
+              hidden: this.tempYear !== year,
+              arrowPosition: true,
+            }}
+            name="chevron-right"
+            size="12"
+          ></ix-icon>
+          <div style={{ 'min-width': 'max-content' }}>{`${year}`}</div>
+        </div>
+      );
+    }
+
+    return rows;
+  }
+
+  render() {
+    return (
+      <Host>
+        <div class="container">
```

Comment:
Same problem as inside time-picker, increaasing the width increase only the host but not the `container`.

![image](https://github.com/siemens/ix/assets/2144313/738794b1-976a-4bd8-a9f9-1764205a72c6)

Reply:
resolved